app/src/main/java/com/github/se/polyfit/ui/screen/LoginScreen.kt

```diff
@@ -24,34 +20,33 @@ import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.graphics.Color
 import androidx.compose.ui.layout.ContentScale
-import androidx.compose.ui.platform.LocalContext
 import androidx.compose.ui.platform.testTag
 import androidx.compose.ui.res.painterResource
 import androidx.compose.ui.unit.dp
 import androidx.compose.ui.unit.sp
+import androidx.hilt.navigation.compose.hiltViewModel
 import com.firebase.ui.auth.FirebaseAuthUIActivityResultContract
-import com.firebase.ui.auth.data.model.FirebaseAuthUIAuthenticationResult
 import com.github.se.polyfit.R
-import com.github.se.polyfit.ui.navigation.Navigation
-import com.google.android.gms.auth.api.signin.GoogleSignIn
-import com.google.android.gms.auth.api.signin.GoogleSignInOptions
+import com.github.se.polyfit.ui.viewModel.LoginViewModel
 
 @Composable
-fun LoginScreen(navController: Navigation) {
-
+fun LoginScreen(goTo: () -> Unit) {
   // Create an instance of the Authentication class
-  val authentication = Authentication(navController, context = LocalContext.current)
+
+  val viewModel: LoginViewModel = hiltViewModel()
+  // val authentication = AuthenticationCloud(context = LocalContext.current){if (it)
+  // navController.navigateToHome()}
   val signInLauncher =
       rememberLauncherForActivityResult(contract = FirebaseAuthUIActivityResultContract()) { res ->
-        authentication.onSignInResult(res)
+        viewModel.onSignInResult(res) { if (it) goTo() else Log.d("LoginScreen", "Sign in failed") }
       }
 
   // Set the signInLauncher in the Authentication class
-  authentication.setSignInLauncher(signInLauncher)
+  viewModel.setSignInLauncher(signInLauncher)
 
   // This function starts the sign-in process
   fun createSignInIntent() {
-    authentication.signIn()
+    viewModel.signIn()
   }
 
   Surface(
```

Comment:
Could be a good idea to have a logo (or placeholder for now)

Reply:
What do you mean by a logo?